### DISCUSSION TITLE
Unexpected EOF from the server

### DISCUSSION THREAD

**Discussion by pavan0437** [getredash/redash#6021] [Q&A]
We recently migrated to new Microsoft Sql Server DB and I see below error when I try test connection by updating server info in DataSource's.

Connection Test Failed:

b'DB-Lib error message 20017, severity 9:\nUnexpected EOF from the server (<New Server Instance>)\nDB-Lib error message 20002, severity 9:\nAdaptive Server connection failed (<New Server Instance>)\n'

I see following logs in PostgreSQL workload

postgresql 02:59:36.87 
postgresql 02:59:36.88 Welcome to the Bitnami postgresql container
postgresql 02:59:36.88 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql
postgresql 02:59:36.88 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql/issues
postgresql 02:59:36.88 Send us your feedback at containers@bitnami.com
postgresql 02:59:36.88 
postgresql 02:59:36.90 INFO  ==> ** Starting PostgreSQL setup **
postgresql 02:59:36.97 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 02:59:36.97 INFO  ==> Loading custom pre-init scripts...
postgresql 02:59:36.98 INFO  ==> Initializing PostgreSQL database...
postgresql 02:59:37.00 INFO  ==> postgresql.conf file not detected. Generating it...
postgresql 02:59:37.01 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 02:59:37.01 INFO  ==> Generating local authentication configuration
postgresql 02:59:37.02 INFO  ==> Deploying PostgreSQL with persisted data...
postgresql 02:59:37.02 INFO  ==> Configuring replication parameters
postgresql 02:59:37.05 INFO  ==> Configuring fsync
postgresql 02:59:37.06 INFO  ==> Loading custom scripts...
postgresql 02:59:37.06 INFO  ==> Enabling remote connections
postgresql 02:59:37.07 INFO  ==> Stopping PostgreSQL...
postgresql 02:59:37.08 INFO  ==> ** PostgreSQL setup finished! **
postgresql 02:59:37.14 INFO  ==> ** Starting PostgreSQL **
LOG:  database system was shut down at 2023-05-02 02:58:44 GMT
LOG:  MultiXact member wraparound protections are now enabled
LOG:  database system is ready to accept connections
LOG:  autovacuum launcher started
LOG:  unexpected EOF on client connection with an open transaction
LOG:  unexpected EOF on client connection with an open transaction
LOG:  unexpected EOF on client connection with an open transaction
LOG:  unexpected EOF on client connection with an open transaction
LOG:  unexpected EOF on client connection with an open transaction
LOG:  unexpected EOF on client connection with an open transaction

Please help

**Comment by pavan0437** [comment]
@susodapop could you please help us here

**Reply by susodapop** [reply]
I'm not contributing to Redash at the moment so I'm a bit out of the loop. But it does appear you're using bitnami images which are not sourced from this repository and have never had very good support. `Adaptive server connection failed` is an extremely generic error. It would probably make sense to SSH into your container and attempt to connect to your database outside the Redash application as this is likely not a problem with Redash source code but rather your configuration.

That's all I can say.
